src/coords.c

```diff
@@ -384,24 +384,42 @@ PyObject* GetCoords(PyArrayObject* arr, int include_z) {
   if (result == NULL) {
     return NULL;
   }
+  if (return_index) {
+    npy_intp dims_ind[1] = {size};
+    index = (PyArrayObject*)PyArray_SimpleNew(1, dims_ind, NPY_INTP);
+    if (index == NULL) {
+      Py_DECREF(result);
```

Comment:
It seems like this should raise an exception similar to `CHECK_ALLOC`?

Reply:
In my experience, the python C API will have set an exception in case it returns NULL. I don't really no a way of testing it here though.